src/ContentGetter/PostsGenerator.php

```diff
@@ -12,62 +12,62 @@
  */
 class PostsGenerator implements ContentGetterContract
 {
-    /**
-     * Instance of the query.
-     *
-     * @var WP_Query
-     */
-    protected $query;
+	/**
+	 * Instance of the query.
+	 *
+	 * @var WP_Query
+	 */
+	protected $query;
 
-    /**
-     * Handles getting the content for the search query.
-     *
-     * @param WP_Query $query Instance of the query.
-     *
-     * @return array
-     */
-    public function getContent(WP_Query $query): array
-    {
-        $this->query = $query;
+	/**
+	 * Handles getting the content for the search query.
+	 *
+	 * @param WP_Query $query Instance of the query.
+	 *
+	 * @return array
+	 */
+	public function getContent(WP_Query $query): array
+	{
+		$this->query = $query;
 
-        return $this->generatePosts($this->getQuantity());
-    }
+		return $this->generatePosts($this->getQuantity());
+	}
 
-    /**
-     * Gets the quantity to be generated.
-     *
-     * @return int
-     */
-    private function getQuantity(): int
-    {
-        if (!isset($this->query->query)) {
-            return 4;
-        }
+	/**
+	 * Gets the quantity to be generated.
+	 *
+	 * @return int
+	 */
+	private function getQuantity(): int
+	{
+		if (!isset($this->query->query)) {
```

Comment:
Maybe a separate concern, but this interface will need to communicate total in the current result set and total pages.

Reply:
Remember here that our `PostsGenerator` is an experimental implementation to merely mock creating posts.  It has a private function to get how many posts to create.

In a real-world application, you would want a different approach as you've indicated.  But here, this approach fits the needs of the educational experiment.

Reply:
Right.